### ISSUE TITLE
LinkHints breaks Proton mail

### ISSUE THREAD

**Issue by akiross** [lydell/LinkHints#99]
### Steps to reproduce

1. Go to mail.proton.me
2. Log in, open your inbox
3. Enable column layout
4. Switching to a different mail will break the preview
5. Trying again with LinkHints disabled yields different results

### Screen shot
![Proton Mail broken](https://github.com/user-attachments/assets/ea2d7179-c856-4c71-9ecc-b6f98997aa70)

### Description

When LinkHints is enabled, Proton Mail will fail and break with a message "Something went wrong. Please refresh the page or try again later.". When the page is reloaded, the selected mail (if any) will load properly, but upon switching email, the error will pop up.

The Developer console shows this kind of messages (related to proton's webpack build):

```
TypeError: can't access dead object
    useLinkHandler.tsx:158
    ...
```
which do not appear when LinkHints is disabled.

### Debug info

<details>
<summary>Debug info</summary>

```json
{
  "version": "1.3.3",
  "browser": "firefox",
  "userAgent": "Mozilla/5.0 (X11; Linux x86_64; rv:130.0) Gecko/20100101 Firefox/130.0",
  "browserInfo": {
    "name": "Firefox",
    "vendor": "Mozilla",
    "version": "130.0.1",
    "buildID": "20240913135723"
  },
  "platformInfo": {
    "os": "linux",
    "arch": "x86-64"
  },
  "storage.sync": {
    "autoActivate": false
  },
  "storage.local": {
    "tutorialShown": true
  },
  "language": "en-US"
}
```

</details>

**Comment by ludams** [comment]
I have the same issue, which is really annoying. Fixing the problem properly would obviously be the best solution for everyone. But in the meantime I created a workaround for myself by disabling the extension for the ProtonMail website by modifying the `src/manifest.ts` file.

I added for both `content_scripts` the property `exclude_matches` with a single entry of `"*://mail.proton.me/*"`. The result looks something like this:

```ts
export default (): string =>
  toJSON({
    // ...
    content_scripts: [
      {
        matches: ["<all_urls>"],
        exclude_matches: [
            "*://mail.proton.me/*",
        ],
        // ...
      },
      {
        matches: ["<all_urls>"],
        exclude_matches: [
            "*://mail.proton.me/*",
        ],
        // ...
      },
    ],
  });
```

And then I built and installed the extension by following the description in the [Contributing documentation](https://github.com/lydell/LinkHints/blob/main/CONTRIBUTING.md).

I hope this helps someone :)

**Comment by n-o-o-b** [comment]
> ### Steps to reproduce
> 
>     1. Go to mail.proton.me
> 
>     2. Log in, open your inbox
> 
>     3. Enable column layout
> 
>     4. Switching to a different mail will break the preview
> 
>     5. Trying again with LinkHints disabled yields different results
> 
> ### Description
> 
> When LinkHints is enabled, Proton Mail will fail and break with a message "Something went wrong. Please refresh the page or try again later.". When the page is reloaded, the selected mail (if any) will load properly, but upon switching email, the error will pop up.

I get this too.

> And then I built and installed the extension by following the description in the [Contributing documentation](https://github.com/lydell/LinkHints/blob/main/CONTRIBUTING.md).

I am not prepared to do this personally, because it involves changing a setting in about:config, which gives me ```Proceed with Caution```

```Changing advanced configuration preferences can impact Firefox performance or security```.

**Comment by n-o-o-b** [comment]
I will have to stop using the extension until this problem is fixed, if ever. It's a shame because this extension is perfect for what I need it for, which is clicking links with the keyboard. It also doesn't require a ton of permissions.

It's shame it's not possible to simply disable it for a specific website, either through the extension's interface or through the Firefox extensions interface.

P.S. I discovered whilst writing this comment [that apparently Mozilla is working on a feature that allows the user to disable add-ons for specific sites](https://connect.mozilla.org/t5/ideas/option-to-allow-extensions-to-run-only-on-specific-websites/idc-p/124597/highlight/true#M64552), so there is hope. It would still be good to get the problem with ProtonMail fixed on the extension level, though. If I could contribute I would.

**Comment by lydell** [comment]
Yeah, it sucks when software you like becomes unusable because of _just that one thing_ :(

I’m not able to fix the Proton issue for the time being.

Thanks for the link about disabling addons per site! It looks like they have implemented the behind-the-scenes code for it, but there’s no UI for actually using it yet – which seems to be tracked here: https://bugzilla.mozilla.org/show_bug.cgi?id=2045795

**Comment by n-o-o-b** [comment]
> I’m not able to fix the Proton issue for the time being.

I understand.

> Thanks for the link about disabling addons per site! It looks like they have implemented the behind-the-scenes code for it, but there’s no UI for actually using it yet – which seems to be tracked here: https://bugzilla.mozilla.org/show_bug.cgi?id=2045795

What's confusing is that the BugZilla ticket the Mozilla employee links to (in the post I linked to) is four years old, but in his post (of 5th of May 2026) he says that they started working on it in the previous week, which would be April or May 2026.

**Comment by arjandepooter** [comment]
Let Kimi3 loose on it and it found the root cause + made a fix. Signed XPI of the fixed version can be found here: https://github.com/arjandepooter/LinkHints/releases/tag/v1.3.3-protonfix.1 . @lydell I created a PR for it but don't know how you feel about AI-generated PRs so feel free to re-open it if you're ok with it.

**Comment by n-o-o-b** [comment]
Thank you!

**Comment by lydell** [comment]
Released in [1.4.0](https://github.com/lydell/LinkHints/issues/1#issuecomment-5374576593). Should get installed in your Firefox shortly.
